tests/trio103.py

```diff
@@ -0,0 +1,90 @@
+import trio
+
+try:
+    pass
+except (SyntaxError, ValueError, BaseException):
+    raise
+except (SyntaxError, ValueError, trio.Cancelled) as p:  # error
+    pass
+except (SyntaxError, ValueError):
+    raise
+except trio.Cancelled as e:
+    raise e
+except:
+    pass
+
+try:
+    pass
+except BaseException:
+    raise ValueError()  # error
+
+
+def foo(var):
+    try:
+        pass
+    except trio.Cancelled:  # error
+        pass
+    except BaseException as e:  # error
+        if True:
+            raise e
+        if True:
+            pass
+        else:
+            raise e
+
+    try:
+        pass
+    except BaseException as e:
+        try:
+            raise e
+        except trio.Cancelled as f:
+            raise f  # ???
+
+    try:
+        pass
+    except BaseException:
+        return  # error
+
+    if True:
+        return
+
+    try:
+        pass
+    except BaseException as e:
+        try:
+            raise e
+        except:
+            raise e
+
+    try:
+        pass
+    except BaseException as e:  # TODO: nested try's
+        try:
+            raise e
+        except:
+            raise
+
+    try:
+        pass
+    except BaseException as e:
+        try:
+            pass
+        except:
+            pass
+        finally:
+            raise e
+
+    try:
+        pass
+    except BaseException as e:
+        raise  # In theory safe? But godawful ugly so treat as error anyway
```

Comment:
No, this is fine, it's actually my preferred style since it doesn't add a frame to the traceback.

Bare `raise` also works whether-or-not you've got an `as` clause on the except handler.

Reply:
Wait, you like bare raises even when you have an `as` clause on the except handler? Bare raises are fine in general
`tests/trio103.py:5`
```python
except (SyntaxError, ValueError, BaseException):
    raise
```
But the case on line 80 is specifically for when you save it in a variable but do a bare raise anyway - in which case you should just skip the `as` clause entirely?
I could create a separate error message for it (~"don't do bare raise when you've saved an exception with an `as` clause"), but a good linter should pick up on the variable `e` being unused.

Reply:
Currently a super-niche case, but see https://peps.python.org/pep-0678/#example-usage

If there really is an unused `as` clause, it should get picked up by a _different_ flake8 plugin, since it's not a trio-related problem 🙂

Reply:
Hmm, I guess that makes sense. I would've written `raise e` in that example, but I guess you don't have to and the bare `raise` is .. shorter. Feels like something a strongly opinionated formatting linter should pick up on.

Also further solidifies the case for not thoroughly parsing nested excepts.

But it's fixed! :+1: